test/celeritas/optical/Cerenkov.test.cc

```diff
@@ -378,6 +392,53 @@ TEST_F(CerenkovTest, TEST_IF_CELERITAS_DOUBLE(generator))
         EXPECT_SOFT_EQ(401.5, avg_engine_samples);
     }
 }
+
+//---------------------------------------------------------------------------//
+
+TEST_F(CerenkovTest, TEST_IF_CELERITAS_DOUBLE(pre_generator))
+{
+    DiagnosticRngEngine<std::mt19937> rng;
+
+    OpticalStepCollectorData step_collector;
+    step_collector.step_length = 0.15 * units::centimeter;
+    step_collector.points[StepPoint::pre].pos = {0, 0, 0};
+    step_collector.points[StepPoint::pre].speed
+        = units::LightSpeed{0.86286196322132458};
+    step_collector.points[StepPoint::post].pos
+        = {step_collector.step_length, 0, 0};
+    step_collector.points[StepPoint::post].speed
+        = units::LightSpeed{0.63431981443206786};
+    CELER_ASSERT(step_collector);
+
+    // 500 keV e-
```

Comment:
You could also test a case where the incident particle speed is below the threshold for Cerenkov production.